macos/FluentUI/Button/Button.swift

```diff
@@ -5,158 +5,254 @@
 	override public var image: NSImage? {
 		willSet {
 			guard wantsLayer == true else {
 				preconditionFailure("wantsLayer must be set so that the image is rendered on the layer")
 			}
 		}
 	}
-	
-	/// String display in the button
+
+	/// Title string to display in the button.
 	override public var title: String {
 		willSet {
 			guard wantsLayer == true else {
 				preconditionFailure("wantsLayer must be set so that the title is rendered on the layer")
 			}
 		}
 	}
-	
-	override public func updateTrackingAreas() {
-		super.updateTrackingAreas()
-		
-		// Remove existing trackingArea
-		if let trackingArea = trackingArea {
-			removeTrackingArea(trackingArea)
-			self.trackingArea = nil
-		}
 
-		// Create a new trackingArea
-		let opts: NSTrackingArea.Options = [.mouseEnteredAndExited, .activeAlways]
-		let trackingArea = NSTrackingArea(
-			rect: bounds,
-			options: opts,
-			owner: self,
-			userInfo: nil
-		)
-		addTrackingArea(trackingArea)
-		self.trackingArea = trackingArea
-	}
+	private var mouseDown = false
 
-	open override func mouseEntered(with event: NSEvent) {
-		mouseEntered = true
-		needsDisplay = true
-	}
+	/// While the current Button is pressed, its style is temporarily applied to the linkedPrimary button.
+	/// This emulates an effect seen in the default style of Cocoa buttons, where pressing a secondary
+	/// button takes the accent color highlighting from a nearby primary button. For best results, the
+	/// current button should have the `.secondary` style, the linkedPrimary button should have the
+	/// `.primary` style, and both buttons should have the same accentColor.
+	@objc public var linkedPrimary: Button?
 
-	open override func mouseExited(with event: NSEvent) {
-		mouseEntered = false
+	private var linkedPrimaryOriginalStyle: ButtonStyle?
+
+	private func setMouse(down: Bool) {
+		mouseDown = down
 		needsDisplay = true
+		if isEnabled {
+			if let linkedPrimary = linkedPrimary
+			{
+				if down {
+					linkedPrimaryOriginalStyle = linkedPrimary.style
+					linkedPrimary.style = self.style
+				} else {
+					linkedPrimary.style = linkedPrimaryOriginalStyle ?? .primary
+				}
+				linkedPrimary.needsDisplay = true
+			}
+		}
 	}
 
 	open override func mouseDown(with event: NSEvent) {
-		mouseDown = true
-		needsDisplay = true
+		setMouse(down: true)
 	}
 
 	open override func mouseUp(with event: NSEvent) {
-		mouseDown = false
-		needsDisplay = true
+		setMouse(down: false)
 	}
 
 	open override func updateLayer() {
 		if let layer = layer {
 			layer.borderWidth = Button.borderWidth
-			layer.cornerRadius = Button.cornerRadius
-			layer.backgroundColor = layerBackgroundColor.cgColor
-			layer.borderColor = outlineColor.cgColor
+			layer.cornerRadius = cornerRadius
+			if let contentTintColor = contentTintColor {
+				super.contentTintColor = contentTintColor
+			}
+			if let backgroundColor = backgroundColor { 
+				layer.backgroundColor = backgroundColor.cgColor
+			}
```

Comment:
do we need to clear the layer's backgroundColor (and borderColor below) if these are nil?

Reply:
Since they are nullable, sure.